src/utils/utils_log.c

```diff
@@ -153,13 +152,21 @@ static void utils_log_internal(utils_log_level_t level, int perror,
             }
             errno = saveno;
 #else
-            char err_buff[1024]; // max size according to manpage.
             int saveno = errno;
             errno = 0;
+#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE
+            char err[1024];
+            int err_ret = strerror_r(saveno, err, sizeof(err));
+            if (err_ret == ERANGE) {
+                postfix = "[truncated...]";
+            }
+#else
```

Comment:
is it diffrent from the apple version? can you just update apple ifdef

Reply:
Done.